nodes/list/get_list_element_index.py

```diff
@@ -0,0 +1,66 @@
+import bpy
+from bpy.props import *
+from ... tree_info import keepNodeLinks
+from ... events import executionCodeChanged
+from ... base_types.node import AnimationNode
+from ... sockets.info import getBaseDataTypeItems, toIdName, toListIdName, isBase, toBaseDataType
+
+class GetListElementIndexNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_GetListElementIndexNode"
+    bl_label = "Get Element Index"    #Search List Element ?
+    
+    def assignedTypeChanged(self, context):
+        self.baseIdName = toIdName(self.assignedType)
+        self.listIdName = toListIdName(self.assignedType)
+        self.generateSockets()
+
+    assignedType = StringProperty(update = assignedTypeChanged)
+    baseIdName = StringProperty()
+    listIdName = StringProperty()
+    
+    def create(self):
+        self.assignedType = "Float"  #could use generic to skip all update but users may try funny stuf (I will..)
+        self.outputs.new("an_IntegerSocket", "First Index", "firstIndex")
+        self.outputs.new("an_IntegerListSocket", "All Indices", "allIndices")
+        self.outputs.new("an_IntegerSocket", "Occurrences", "occurrences")
+        
+    def drawAdvanced(self, layout):
+        self.invokeSocketTypeChooser(layout, "assignListDataType",
+            socketGroup = "LIST", text = "Change Type", icon = "TRIA_RIGHT")
+        
+    def getExecutionCode(self):
+        isLinked = self.getLinkedOutputsDict()
+        if not any(isLinked.values()): return ""
+        
+        lines = []
+        lines.append("allIndices = [i for i, element in enumerate(list) if element == search]")
+        if isLinked["firstIndex"]: lines.append("firstIndex = allIndices[0] if len(allIndices) > 0 else None")  #this may be the most used, so we just avoid another get element
```

Comment:
the firstIndex must be an integer. Assigning *None* to the variable can cause hard to detect bugs in other places because all other nodes rely on the fact that an integer socket always gets an integer value.
Many programming languages return -1 as index when the element was not found. i think we should do the same here.